src/middlewares/checkAdminApiHeaders.ts

```
@@ -56,3 +61,38 @@ export default function checkAdminApiVersionHeaders(
 
   next()
 }
+
+/**
+ * A middleware asserting that all Admin PATCH API HTTP requests have an appropriate Content-Type header.
+ *
+ * @param req - An Express Request object.
+ * @param res - An Express Response object.
+ * @param next - An Express next() function.
+ *
+ * @throws A ParseError if the Content-Type header is missing, malformed, or unsupported.
+ */
+export function checkAdminApiPatchHeaders(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+): void {
+  // The merge patch format is primarily intended for use with the HTTP PATCH method
+  // as a means of describing a set of modifications to a target resource’s content.
+  // application/merge-patch+json is a Type Specific Variation of the "application/merge-patch" Media Type that uses a
+  // JSON data structure to describe the changes to be made to a target resource.
+  const patchContentType = 'application/merge-patch+json'
+
+  // Add this header to all successful responses. We add it early so even errors will respond with Accept-Patch header.
+  // The Accept-Patch response HTTP header advertises which media-type the server
+  // is able to understand while receiving a PATCH request.
+  res.header('Accept-Patch', patchContentType)
+
+  if (req.header('Content-Type') !== patchContentType) {
+    throw new HeaderError(
+      `A specific 'Content-Type' header is required for a PATCH request: 'Content-Type: ${patchContentType}'.`,
+      HeaderErrorType.UnsupportedMediaTypeHeader,
```

You might be able to add this in the base `HeaderError` constructor. Are there multiple `HeaderErrorType`s that you anticipate?

---

After searching on the web, I don't think we will need another `HeaderErrorTypes` so I moved it to the constructor directly.

I changed the constructor by including the possibility to throw an error for requests omitting `Content-Type: application/json`. This will be useful for the `POST` and `PUT` endpoints.
I did a test using Postman and at the moment if you try to create a PayID with the correct payload (including a `payId` field) but without `Content-Type`, you will see the following `400` error:
```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "A `payId` must be provided in the request body."
}
```
which is not the best in my opinion. We should return a `415` in this case. RippleNet also returns a `415` in such cases.
So now with the updated `checkAdminApiContentTypeHeaders` function, it will be possible to do that check.

What do you think?

---

That's a great observation! Would you make a GitHub issue that captures that we should throw a 415 for POST and PUT when given an invalid Content-Type header?

That can be a follow-on PR after this gets merged.